fdbserver/BlobManager.actor.cpp

```diff
@@ -5628,92 +5628,100 @@ ACTOR Future<bool> shouldBackupManifest(Reference<BlobManagerData> bmData) {
-			}
+	// Wait for next manifest dumped
+	state int64_t seqNo = bmData->manifestDumperSeqNo;
+	loop {
+		TraceEvent("WaitingBlobManifest").detail("Seq", seqNo);
+		wait(bmData->manifestCompletitionTrigger.onTrigger());
+		if (bmData->manifestDumperSeqNo > seqNo) {
+			TraceEvent("BlobManifestDumped").detail("Seq", bmData->manifestDumperSeqNo);
+			break;
+		}
+	}
 
-			// Truncate mutation logs to max retention period
-			Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(bmData->db));
-			Optional<int64_t> logEndEpochs = wait(timeKeeperEpochsFromVersion(logEndVersion, tr));
-			if (!logEndEpochs.present()) {
-				TraceEvent("SkipMutationLogTruncation").detail("LogEndVersion", logEndVersion);
-				return Void(); // skip truncation if no timestamp about log end
-			}
+	// Truncate mutation logs to max retention period
+	Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(bmData->db));
+	Optional<int64_t> logEndEpochs = wait(timeKeeperEpochsFromVersion(logEndVersion, tr));
+	if (!logEndEpochs.present()) {
+		TraceEvent("SkipMutationLogTruncation").detail("LogEndVersion", logEndVersion);
+		return Void(); // skip truncation if no timestamp about log end
+	}
 
-			// Find timestamp and version to truncate
-			int64_t epochs = logEndEpochs.get() - SERVER_KNOBS->BLOB_RESTORE_MLOGS_RETENTION_SECS;
-			if (epochs <= 0) {
-				TraceEvent("SkipMutationLogTruncation").detail("Epochs", epochs);
-				return Void();
-			}
-			state std::string timestamp = BackupAgentBase::formatTime(epochs);
-			state Version truncVersion = wait(timeKeeperVersionFromDatetime(timestamp, bmData->db));
+	// Find timestamp and version to truncate
+	int64_t epochs = logEndEpochs.get() - SERVER_KNOBS->BLOB_RESTORE_MLOGS_RETENTION_SECS;
+	if (epochs <= 0) {
+		TraceEvent("SkipMutationLogTruncation").detail("Epochs", epochs);
+		return Void();
+	}
+	state std::string timestamp = BackupAgentBase::formatTime(epochs);
+	state Version truncVersion = wait(timeKeeperVersionFromDatetime(timestamp, bmData->db));
+
+	wait(bc->expireData(truncVersion, true));
+	bmData->stats.lastMLogTruncationVersion = truncVersion;
+	TraceEvent("TruncateMutationLogs").detail("Version", truncVersion).detail("Timestamp", timestamp);
+	CODE_PROBE(true, "Flush blob granules and truncate mutation logs");
+	return Void();
+}
 
-			wait(bc->expireData(truncVersion, true));
-			bmData->stats.lastMLogTruncationVersion = truncVersion;
-			TraceEvent("TruncateMutationLogs").detail("Version", truncVersion).detail("Timestamp", timestamp);
-			CODE_PROBE(true, "Flush blob granules and truncate mutation logs");
+ACTOR Future<Void> truncateMutationLogsLoop(Reference<BlobManagerData> bmData) {
+	loop {
+		try {
+			wait(truncateMutationLogs(bmData));
 		} catch (Error& e) {
+			if (e.code() == error_code_operation_cancelled) {
+				throw;
+			}
 			TraceEvent("TruncateMutationLogsError").error(e); // skip and retry next time
 		}
+		wait(delay(SERVER_KNOBS->BLOB_MANIFEST_BACKUP_INTERVAL));
```

Comment:
Rhis isn't strictly related to this PR, but because this wait loop isn't durable, this means every time we bounce the cluster or the blob manager, we will flush the granules and truncate the mutation logs, right?
Because this operation is expensive and thus intended to be infrequent, we should probably make this durable somehow so that we actually flush on the intended interval across restarts

Reply:
> every time we bounce the cluster or the blob manager, we will flush the granules and truncate the mutation logs
We don't do that for every restart. It checks last flush time and won't flush/truncate if it has done that BLOB_RESTORE_MLOGS_RETENTION_SECS ago.. so it's already durable. 

One related change is to move last flush time into BlobGranulesBackupConfig so that all backup/restore config are managed in same place.

Reply:
oh right this is just the manifest backup interval not the flush interval, my bad